### PR TITLE
visualize environment

### PR DESCRIPTION
**Pull request by nevio18324**
…ng environment

### CODE REVIEW

File: backend/src/main/resources/application-demo.properties [hunk 1]
```diff
@@ -7,15 +7,19 @@ spring.security.oauth2.resourceserver.opaquetoken.client-id=pitc_okr_demo
...
 okr.tenants.pitc.clientcustomization.favicon=assets/favicon.png
 okr.tenants.pitc.clientcustomization.title=Puzzle OKR
+okr.tenants.pitc.clientcustomization.helpSiteUrl=https://wiki.puzzle.ch/Puzzle/OKRs
 okr.tenants.pitc.clientcustomization.customstyles.okr-topbar-background-color=#1e5a96
 okr.tenants.pitc.clientcustomization.customstyles.okr-banner-background-color=#dcedf9
-okr.tenants.pitc.clientcustomization.helpSiteUrl=https://wiki.puzzle.ch/Puzzle/OKRs
-
+okr.tenants.pitc.clientcustomization.customstyles.okr-overview-background-color=#003153
+okr.tenants.pitc.clientcustomization.customstyles.okr-team-header-color=#ffffff
+okr.tenants.pitc.clientcustomization.customstyles.okr-add-objective-text-color=#ffffff
```
Comment (RandomTannenbaum) [inline backend/src/main/resources/application-demo.properties:20]: Use mid-grey (see figma)

File: backend/src/main/resources/application-demo.properties [hunk 2]
```diff
@@ -7,15 +7,19 @@ spring.security.oauth2.resourceserver.opaquetoken.client-id=pitc_okr_demo
...
+okr.tenants.pitc.clientcustomization.customstyles.okr-add-objective-icon=/assets/icons/new-icon-demo.svg
+okr.tenants.pitc.clientcustomization.customstyles.okr-add-objective-outline-color=#ffffff
```
Comment (RandomTannenbaum) [inline backend/src/main/resources/application-demo.properties:22]: Check outline color

File: frontend/src/app/components/team/team.component.ts [hunk 3]
```diff
@@ -16,16 +19,20 @@ import { DialogService } from '../../services/dialog.service';
 export class TeamComponent implements OnInit {
   @Input({ required: true })
   public overviewEntity!: OverviewEntity;
+  protected addIconSrc: Observable<string> = this.configService.config$.pipe(
```
Comment (RandomTannenbaum) [inline frontend/src/app/components/team/team.component.ts:22]: Maybe rewrite to using behaviorsubject. For example see overview.component.ts

File: frontend/src/style/styles.scss [hunk 4]
```diff
@@ -121,7 +121,7 @@ body.okr {
 }
 
 .bg-grey {
```
Comment (RandomTannenbaum) [inline frontend/src/style/styles.scss:123]: Check css class naming

**Review by RandomTannenbaum — APPROVED** [review]

File: frontend/src/app/components/team/team.component.ts [hunk 5]
```diff
@@ -16,16 +19,22 @@ import { DialogService } from '../../services/dialog.service';
...
   constructor(
     private dialogService: DialogService,
     private refreshDataService: RefreshDataService,
-  ) {}
-
-  trackByObjectiveId: TrackByFunction<ObjectiveMin> = (index, objective) => objective.id;
+    private configService: ConfigService,
+  ) {
+    this.configService.config$.subscribe((config: ClientConfig) => {
+      this.addIconSrc.next(config.customStyles['okr-add-objective-icon'] ?? '/assets/icons/new-icon.svg');
```
Comment (MasterEvarior) [inline frontend/src/app/components/team/team.component.ts:30]:
```suggestion
      this.addIconSrc.next(config.customStyles['okr-add-objective-icon']);
```
I think this  is redundant, as the `BehaviorSubject` already has the same default value :thinking: 
I'd rather have a check if the `customStyles[...]` value is `null` and only changing it if it isn't.

File: frontend/src/app/components/team/team.component.ts
Comment (MasterEvarior) [inline frontend/src/app/components/team/team.component.ts:1]: Maybe also a 1-2 simple unit tests:
- one to check that the correct icon is set
- one the check that the default is applied if nothing is set

**Review by MasterEvarior — CHANGES_REQUESTED** [review]
Almost there :+1:

File: frontend/src/app/components/team/team.component.spec.ts [hunk 6]
```diff
@@ -85,4 +85,11 @@ describe('TeamComponent', () => {
     const button = fixture.debugElement.query(By.css('[data-testId="add-objective"]'));
     expect(button).toBeFalsy();
   });
+
+  it('should display right add objective icon', () => {
```
Comment (MasterEvarior) [inline frontend/src/app/components/team/team.component.spec.ts:89]:
```suggestion
  it('should display correct add objective icon', () => {
```
Nitpick

File: frontend/src/app/components/team/team.component.html [hunk 7]
```diff
@@ -1,18 +1,23 @@
...
       color="primary"
       mat-button
     >
-      <span class="d-flex align-items-center add-text">
-        <img alt="Add key-result button" class="add-cross-button" src="../../../assets/icons/new-icon.svg" />
+      <span id="add-objective-span" class="d-flex align-items-center add-text">
+        <img
+          alt="Add key-result button"
+          class="add-cross-button"
+          src="{{ (addIconSrc | async) ?? '/assets/icons/new-icon.svg' }}"
```
Comment (MasterEvarior) [inline frontend/src/app/components/team/team.component.html:19]:
```suggestion
          src="{{ (addIconSrc | async)  }}"
```
Remove from here ...

File: frontend/src/app/components/team/team.component.ts [hunk 8]
```diff
@@ -16,16 +19,22 @@ import { DialogService } from '../../services/dialog.service';
...
     private dialogService: DialogService,
     private refreshDataService: RefreshDataService,
-  ) {}
-
-  trackByObjectiveId: TrackByFunction<ObjectiveMin> = (index, objective) => objective.id;
+    private configService: ConfigService,
+  ) {
+    this.configService.config$.subscribe((config: ClientConfig) => {
+      this.addIconSrc.next(config.customStyles['okr-add-objective-icon']);
+    });
```
Comment (MasterEvarior) [inline frontend/src/app/components/team/team.component.ts:31]:
```suggestion
    this.configService.config$.subscribe((config: ClientConfig) => {
      if(config.customStyles['okr-add-objective-icon']){
         this.addIconSrc.next(config.customStyles['okr-add-objective-icon']);
      }
    });
```
... and move it down here.

This has the benefit that there is only a single place where the fallback is set.

**Review by MasterEvarior — CHANGES_REQUESTED** [review]

File: backend/src/main/resources/application-demo.properties [hunk 9]
```diff
@@ -1,22 +1,22 @@
...
 okr.tenants.pitc.clientcustomization.favicon=assets/favicon.png
 okr.tenants.pitc.clientcustomization.title=Puzzle OKR
-okr.tenants.pitc.clientcustomization.customstyles.okr-topbar-background-color=#1e5a96
-okr.tenants.pitc.clientcustomization.customstyles.okr-banner-background-color=#dcedf9
 okr.tenants.pitc.clientcustomization.helpSiteUrl=https://wiki.puzzle.ch/Puzzle/OKRs
-
-
+okr.tenants.pitc.clientcustomization.customstyles.okr-topbar-background-color=#1e5a96
+okr.tenants.pitc.clientcustomization.customstyles.okr-banner-background-color=linear-gradient(270deg, #FFFFFF -11.04%, #BEDEF6 40.13%)
+okr.tenants.pitc.clientcustomization.customstyles.okr-overview-background-color=linear-gradient(0deg, #1E5A96, #1E5A96),linear-gradient(180deg, rgba(27, 45, 83, 0.6) 0%, rgba(30, 90, 150, 0.6) 30.84%)
```
Comment (peggimann) [inline backend/src/main/resources/application-demo.properties:15]: somehow the gradient is not applied

File: frontend/src/app/components/team/team.component.ts [hunk 10]
```diff
@@ -16,16 +19,23 @@ import { DialogService } from '../../services/dialog.service';
...
 
   constructor(
     private dialogService: DialogService,
     private refreshDataService: RefreshDataService,
-  ) {}
-
-  trackByObjectiveId: TrackByFunction<ObjectiveMin> = (index, objective) => objective.id;
+    private configService: ConfigService,
+  ) {
+    this.configService.config$.subscribe((config: ClientConfig) => {
```
Comment (peggimann) [inline frontend/src/app/components/team/team.component.ts:29]: this subscribe poses a potential memory leak because it is never properly destroyed

**Review by peggimann — CHANGES_REQUESTED** [review]

File: frontend/src/app/components/team/team.component.html [hunk 11]
```diff
@@ -1,18 +1,24 @@
-<div *ngIf="overviewEntity as OVEntity" class="bg-grey justify-content-between pt-3">
```
Comment (ManuelMoeri) [inline frontend/src/app/components/team/team.component.html:1]: Why was this removed? :thinking:

**Review by ManuelMoeri — CHANGES_REQUESTED** [review]
Make sure to resolve the conversations from before aswell

**Review by ManuelMoeri — APPROVED** [review]
LGTM.

File: backend/src/main/resources/application-demo.properties [hunk 12]
```diff
@@ -1,22 +1,22 @@
...
 spring.flyway.locations=classpath:db/migration,classpath:db/data-migration-demo,classpath:db/callback
-
 # pit client customization
-okr.tenants.pitc.clientcustomization.logo=assets/images/okr-logo.svg
+okr.tenants.pitc.clientcustomization.logo=assets/images/okr-demo-logo.svg
 okr.tenants.pitc.clientcustomization.triangles=assets/images/triangles-okr-header.svg
 okr.tenants.pitc.clientcustomization.background-logo=assets/images/puzzle-p.svg
 okr.tenants.pitc.clientcustomization.favicon=assets/favicon.png
 okr.tenants.pitc.clientcustomization.title=Puzzle OKR
+okr.tenants.pitc.clientcustomization.helpSiteUrl=https://wiki.puzzle.ch/Puzzle/OKRs
```
[the remark this answers is not in the document]
Comment (nevio18324) [reply backend/src/main/resources/application-demo.properties:12]: yes

**Review by MasterEvarior — APPROVED** [review]
Looks good to me on demo and staging :+1:

**Pull request merged**